proximal/examples/test_deconv_sinv_psf.py

```diff
@@ -14,28 +14,42 @@
 from scipy.stats import multivariate_normal
 
 # Create Synthetic Horizontal Stripes Pattern Image
-im_s = np.zeros((2748, 3840))
-stripe_width, stripe_gap, stripe_start, stripe_end = 50, 50, 500, 500
-for y in range(0, im_s.shape[0]-(stripe_start+stripe_end), stripe_width + stripe_gap):
-    im_s[stripe_start+y : stripe_start+y + 50, :] = .8
-
-xx, yy = np.mgrid[0:im_s.shape[0], 0:im_s.shape[1]]
-im_ctr = (np.array(im_s.shape)/2).astype(int)
-r = np.sqrt((xx - im_ctr[0])**2 + (yy - im_ctr[1])**2)
-mask = np.zeros_like(im_s)
-mask[r < 1000] = 1
-im_s *= mask
+def get_raw(width=3840, height=2748):
+    im_s = np.zeros((2748, 3840))
+    stripe_width, stripe_gap, stripe_start, stripe_end = 50, 50, 500, 500
+    for y in range(0, im_s.shape[0]-(stripe_start+stripe_end), stripe_width + stripe_gap):
+        im_s[stripe_start+y : stripe_start+y + 50, :] = .8
+
+    xx, yy = np.mgrid[0:im_s.shape[0], 0:im_s.shape[1]]
+    im_ctr = (np.array(im_s.shape)/2).astype(int)
+    r = np.sqrt((xx - im_ctr[0])**2 + (yy - im_ctr[1])**2)
+    mask = np.zeros_like(im_s)
+    mask[r < 1000] = 1
+    im_s *= mask
+
+    return np.asfortranarray(im_s, dtype=np.float32), im_ctr
+
+im_s, im_ctr = get_raw()
 plt.subplot(221); plt.imshow(im_s); plt.title('Ground Truth');
 
 # Create Gaussian Kernel & Convolve
-x, y = np.mgrid[0:im_s.shape[0], 0:im_s.shape[1]]
-pos = np.dstack((x, y))
-rv = multivariate_normal.pdf(pos, im_ctr, [[200, 0], [300, 500]])
-psf2 = rv/np.max(rv)
-psf2_cropped = psf2[im_ctr[0]-100:im_ctr[0]+101, im_ctr[1]-100:im_ctr[1]+101]
-plt.subplot(222); plt.imshow(psf2_cropped); plt.title('PSF zoomed in');
-
-im_sb = fftconvolve(im_s, psf2, mode='same')
+def get_blur_kernel(im_shape, im_ctr, crop=True):
+    x, y = np.mgrid[0:im_shape[0], 0:im_shape[1]]
+    pos = np.dstack((x, y))
+    rv = multivariate_normal.pdf(pos, im_ctr, [[200, 0], [300, 500]])
+    psf2 = np.asfortranarray(rv/np.max(rv), dtype=np.float32)
+
+    if not crop:
+        return psf2
+
+    psf2 = psf2[im_ctr[0]-100:im_ctr[0]+101, im_ctr[1]-100:im_ctr[1]+101]
+
+    return psf2
```

Comment:
I added an option to crop and fit to the size of the PSF. Cropping is combined with `scipy.signal.convolve(... , method='auto')` to accelerate the 2D convolution.